azurelinuxagent/common/osutil/clearlinux.py

```diff
@@ -75,18 +76,14 @@ def conf_sshd(self, disable_password):
     def del_root_password(self):
         try:
             passwd_file_path = conf.get_passwd_file_path()
-            try:
-                passwd_content = fileutil.read_file(passwd_file_path)
-                if not passwd_content:
-                    # Empty file is no better than no file
-                    raise FileNotFoundError # pylint: disable=undefined-variable
-            except FileNotFoundError: # pylint: disable=undefined-variable
+            passwd_content = fileutil.read_file(passwd_file_path)
+            if not passwd_content:
                 new_passwd = ["root:*LOCK*:14600::::::"]
```

Comment:
Ah. Looking at this again, I'm seeing that this isn't correct. I didn't realize that `fileutil.read_file` could also throw a `FileNotFoundError`... The problem here is that python2 does not have that builtin.

Should `read_file` throw an error that works with python2?